homeassistant/components/stream/__init__.py

```diff
@@ -113,41 +116,44 @@ def shutdown(event):
 class Stream:
     """Represents a single stream."""
 
-    def __init__(self, hass, source, options=None):
+    def __init__(
+        self, hass: HomeAssistant, source: str, options: dict[str, str]
+    ) -> None:
         """Initialize a stream."""
         self.hass = hass
         self.source = source
         self.options = options
         self.keepalive = False
-        self.access_token = None
-        self._thread = None
+        self.access_token: str | None = None
+        self._thread: threading.Thread | None = None
         self._thread_quit = threading.Event()
         self._outputs: dict[str, StreamOutput] = {}
         self._fast_restart_once = False
 
-        if self.options is None:
-            self.options = {}
-
     def endpoint_url(self, fmt: str) -> str:
         """Start the stream and returns a url for the output format."""
         if fmt not in self._outputs:
             raise ValueError(f"Stream is not configured for format '{fmt}'")
         if not self.access_token:
             self.access_token = secrets.token_hex()
-        return self.hass.data[DOMAIN][ATTR_ENDPOINTS][fmt].format(self.access_token)
+        return cast(
+            str, self.hass.data[DOMAIN][ATTR_ENDPOINTS][fmt].format(self.access_token)
+        )
```

Comment:
I think this is a bit cleaner.
```suggestion
        endpoint_fmt: str = self.hass.data[DOMAIN][ATTR_ENDPOINTS][fmt]
        return endpoint_fmt.format(self.access_token)
```